Notebooks that serve as examples for each model template

karthikmswamy opened the pull request:
Including notebook examples for generic, hierarchical and regression model templates.

**Checks**:

- [x] README.md is maintained
- [x] [Documentation](/docs) is maintained: not relevant for this PR
- [x] CHANGELOG.md is updated: will follow up with a new commit
- [x] Tests added: does not seem relevant to this PR

misch commented on notebooks/Data_Attribute_Recommendation_Hierarchical_Model_Template.ipynb:
```diff
@@ -0,0 +1,1602 @@
+        "id": "K2QywZBDzE1o"
+      },
+      "source": [
+        "Once you downloaded the service key from the Cockpit, upload it to your notebook environment. The service key must be uploaded to same directory where the `teched2020-INT260_Data_Attribute_Recommendation.ipynb` is stored.\n",
```
```suggestion
        "Once you downloaded the service key from the Cockpit, upload it to your notebook environment. The service key must be uploaded to same directory where the `Data_Attribute_Recommendation_Hierarchical_Model_Template.ipynb` is stored.\n",
```

karthikmswamy replied:
Fixed this with the correct name now.

misch commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
```suggestion
    "Once you downloaded the service key from the Cockpit, upload it to your notebook environment. The service key must be uploaded to same directory where the `Data_Attribute_Recommendation_Regression_Model_Template.ipynb` is stored.\n",
```

mhaas replied:
Yep, we probably need to update the file name!

karthikmswamy replied:
Fixed all occurrences of TechEd to point to the file name.

misch commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
I generally like to see the outputs of `read_model_collection`, `read_job_collection` etc., and perhaps we can move those bits to the places right after the respective things are created? Also, a bit of "explanation text" could be helpful

mhaas replied:
Interesting point! For last year's workshop, I had opted to keep the `read_*_collection` methods as "optional exercises" towards the end of the notebook. They are also in this notebook towards the end.

I did not have the various methods in the main flow because I wanted to focus a bit on the story and not so much on all the details. I think that is good for a workshop. Maybe for self-study offline outside of a workshop, it would be OK to have more details at each stage.

I am fine with either way, but if we want to use the notebooks also for a workshop, I would prefer keeping the things optional. Or we can make a single example at the beginning for the dataset schema for `get_dataset_schema_by_id` and `read_dataset_schemas_collection` and state that there are similar methods for the other resources.

karthikmswamy replied:
I agree that this might be useful but seldom do we encounter someone reading all the text in the notebook. Most users simply run the notebooks to see the outputs and get to their results as quickly as possible. We can keep this as a way forward to improve the notebooks. I've added an issue linking to this comment so that we have the context in the future: https://github.com/SAP/data-attribute-recommendation-python-sdk/issues/119

misch commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
is the `top_n` parameter relevant for the regression model?

mhaas replied:
I think this is still for the hierarchical template. We could think of removing this because it is redundant with the actual model template used in the code section below.

mhaas replied:
Is it a label?

Maybe this would be more accurate:

```suggestion
    "* Getting predictions for new data\n",
```

Still not great, though.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1534 @@
+        "# ! pip install pandas\n",
+        "import pandas as pd\n",
+        "\n",
+        "df = pd.read_csv(\"./arxiv.csv.gz\")\n",
```
Thanks for bringing this up. I'm trying to get a good source to make sure we don't store the data in this repo. I'll update this with a `wget` line once I find a good source.

karthikmswamy replied:
I've updated this to include the data in a separate folder along with the license for its usage. Let me know if this looks better now.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<video controls src=\"videos/dar_prediction_material_table.mp4\"/>\n",
```
I think the video is missing. I am not sure if we should bring it over: at least for the regression template, it might be confusing.

karthikmswamy replied:
Yes, I've included the video only for the hierarchical template now.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+   "source": [
+    "## Business Scenario\n",
+    "\n",
+    "We will consider a business scenario involving product master data. The creation and maintenance of this product master data requires the careful manual selection of the correct categories for a given product from a pre-defined hierarchy of product categories.\n",
```
I think the introduction here still talks about the master data use case where we predict categories. Can you update that to reference the price?

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Please log in to your trial account: https://cockpit.eu10.hana.ondemand.com/trial/\n",
```
We may have to update this URL to US10. I am currently creating my trial account from scratch so I will check later.

karthikmswamy replied:
Done. Followed your regression notebook to get this in line.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+    "\n",
+    "![trial_booster.png](attachment:trial_booster.png)\n",
+    "\n",
+    "*Boosters are only available on the Trial landscape. If you are using a production environment, please follow this tutorial to manually [create a service instance and a service key](https://developers.sap.com/tutorials/cp-aibus-dar-service-instance.html)*."
```
I am afraid that US10 will not have the boosters right now. I will also check this later.

karthikmswamy replied:
Does this still hold true? I believe we have booster for US10 as well, right?

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
We can also update the screenshot because it still shows teched 2020, but it is not that critical.

karthikmswamy replied:
Done.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Agree!

karthikmswamy replied:
Agreed, I've used the `default_key.json` similar in all the notebooks as this would be the file name when one runs the booster. This should be consistent now.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Can you remove all the output cells? That way, there are no spoilers :)

karthikmswamy replied:
Yep, done now.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Since we have a regression use case, maybe we should reduce the focus on the hierarchy here and just delete this part?

karthikmswamy replied:
Agree, updated the notebook to be the one you presented at TechEd with the name removed, of course :)

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
```suggestion
    "The Data Attribute Recommendation service supports different ModelTemplates:\n",
```

karthikmswamy replied:
Fixed to use your latest regression notebook.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Maybe this would be better?

```suggestion
    "The job is now running in the background. Similar to the Dataset validation, we have to poll the job until it succeeds.\n",
```

karthikmswamy replied:
Fixed to use your latest regression notebook.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
It would be great to have a few sentences here how to interpret these. For example, that we need to consider the `meanAbsoluteError` in the context of the range of data (can we get that from the dataframe?). And maybe how to interpret the `meanAbsolutePercentageError`: `239%` seems quite bad, but maybe it is not?

karthikmswamy replied:
Fixed to use your latest regression notebook.

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Do you happen to have an example where the results are closer to reality?

Maybe it would also be good to explain in one sentence what the `std` value means and how it is computed.

karthikmswamy replied:
Fixed to use your latest regression notebook. You've already followed up with this with a new dataset - so we're covered here :)

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Can you also update this example?

karthikmswamy replied:
Fixed to use your latest regression notebook.

mhaas reviewed (commented):
Thanks for the notebooks! This is super useful. I added a few comments. We can also have a sync this week to check how I can help out!

mhaas commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1631 @@
+        "\n",
+        "In the your global account screen, go to the \"Boosters\" tab:\n",
+        "\n",
+        "![trial_booster.png](attachment:trial_booster.png)\n",
```
I think the attachments got lost during the cleanup. The screenshots were present in the initial commit.

karthikmswamy replied:
I've now included the screenshots as file paths. I'm unable to add images as attachments as you did. We need to see how you did that and replicate in all the notebooks.

mhaas reviewed (changes requested):
Can you bring back the screenshots?

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1534 @@
+      "source": [
+        "**In the file browser, navigate to the directory where the `teched2020-INT260_Data_Attribute_Recommendation.ipynb` notebook file is stored. The service key must reside next to this file.**\n",
+        "\n",
+        "In the Jupyter file browser, click the **Upload** button (1). In the file selection dialog that opens, select the `defaultKey_*.json` file you downloaded previously from the SAP Cloud Platform Cockpit. Rename the file to `key.json`. \n",
```
Done with the latest changes.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1534 @@
+        "- QBI: Quantitative Biology\n",
+        "- PHY: Physics\n",
+        "\n",
+        "The first title, has the following set of labels:\n",
```
Done. Thanks for this suggestion :)

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1534 @@
+      "source": [
+        "We first have to describe the shape of our data by creating a DatasetSchema. This schema informs the service about the individual column types found in the CSV. We also describe which are the target columns used for training. These columns will be later predicted.\n",
+        "\n",
+        "The service currently supports three column types: **text**, **category** and **number**. For prediction, only **category** is currently supported.\n",
```
You're right. I missed out see this. I've fixed it now.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1534 @@
+        "id": "poGMX3K4Kpfm"
+      },
+      "source": [
+        "The [`DataManagerClient`](https://data-attribute-recommendation-python-sdk.readthedocs.io/en/latest/api.html#sap.aibus.dar.client.data_manager_client.DataManagerClient) class is also responsible for uploading data to the service. This data must fit to an existing DatasetSchema. After uploading the data, the service will validate the Dataset against the DataSetSchema in a background proce ...
```
Fixed it in all the notebooks now. Thanks for finding this.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
````diff
@@ -0,0 +1,1534 @@
+        "}\n",
+        "```\n",
+        "\n",
+        "Refer to the [official documentation on ModelTemplates](https://help.sap.com/viewer/105bcfd88921418e8c29b24a7a402ec3/SHIP/en-US/1e76e8c636974a06967552c05d40e066.html) to learn more. Additional model templates may be added over time, so check back regularly."
````
Fixed it in all the notebooks with the list of model templates. Thanks for finding this.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1534 @@
+        "\n",
+        "In the your global account screen, go to the \"Boosters\" tab:\n",
+        "\n",
+        "![trial_booster.png](attachment:trial_booster.png)\n",
```
the paths for all the images have now been properly routed. Somehow, the PR still does not reflect the images in the notebook.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Hierarchical_Model_Template.ipynb:
```diff
@@ -0,0 +1,1602 @@
+        "id": "Chs1mue3zE1g"
+      },
+      "source": [
+        "<video controls src=\"videos/dar_prediction_material_table.mp4\"/>\n",
```
Updated the video into the `videos` folder along with the updated path.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Hierarchical_Model_Template.ipynb:
```diff
@@ -0,0 +1,1602 @@
+        "id": "0pkAx7nGzE1o"
+      },
+      "source": [
+        "**In the file browser, navigate to the directory where the `teched2020-INT260_Data_Attribute_Recommendation.ipynb` notebook file is stored. The service key must reside next to this file.**\n",
```
Fixed this with the correct name.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Hierarchical_Model_Template.ipynb:
```diff
@@ -0,0 +1,1602 @@
+      "source": [
+        "We first have to describe the shape of our data by creating a DatasetSchema. This schema informs the service about the individual column types found in the CSV. We also describe which are the target columns used for training. These columns will be later predicted. In our case, these are the three category columns.\n",
+        "\n",
+        "The service currently supports three column types: **text**, **category** and **number**. For prediction, only **category** is currently supported.\n",
```
Fixed this with the proper sentence.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Hierarchical_Model_Template.ipynb:
```diff
@@ -0,0 +1,1602 @@
+        "\n",
+        "The first defined label is `level1_category`, which is given more weight during training than `level3_category`.\n",
+        "\n",
+        "Refer to the [official documentation on ModelTemplates](https://help.sap.com/viewer/105bcfd88921418e8c29b24a7a402ec3/SHIP/en-US/1e76e8c636974a06967552c05d40e066.html) to learn more. Additional model templates may be added over time, so check back regularly."
```
Invariably, users might run the notebook independently. We need to find a way to get our notebooks to inherit from a base notebook to make things easier to maintain :D

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
This as well

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
This has now been fixed as well. all the notebook contents have been updated to reflect the business case that they deal with.

karthikmswamy commented on notebooks/Data_Attribute_Recommendation_Regression_Model_Template.ipynb:
```diff
@@ -0,0 +1,1933 @@
+  {
+   "attachments": {
+    "trial_booster_finished.png": {
+     "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0oAAAH+CAYAAACiHv6nAAAK0mlDQ1BJQ0MgUHJvZmlsZQAASImVlwdUk8kWx+f70hstgICU0DvSCSAl9ABKr6ISkkBCiTEhqNiVRQXXgooIqAuyIKLgqhRZVMSChUWwgH2DiIC6igUbKvsBj7C777z3zrvnzLm/3Ny5c2fOzDn/DwCKJ0soTIcVAMgQZIrC/b1osXHxNNxTQAKKAAdUAJbFFgsZoaHBALFp/3f70AOgCX/LcqLWv///X02JwxWzAYASEE7iiNkZCLci4ylbKMoEAFWBxPWXZQon+ArCyiKkQYQfTnDKFI9McNIko9GTOZHh3girAYAns1iiFADIBkiclsVOQeqQfRC2 ...
```
Fixed.

karthikmswamy commented:
Hi @misch, thanks for all the wonderful, detailed comments and for taking time to review the notebooks. I've addressed all your comments in this PR. I'm still unable to attach the screenshot as an attachment but all of them are referred by a relative path. I've created an issue to follow up on this in the future: https://github.com/SAP/data-attribute-recommendation-python-sdk/issues/120

Hi @mhaas, I've addressed all your comments as well. Hopefully, the current version looks in a much better state.

misch commented on notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb:
```diff
@@ -0,0 +1,1524 @@
+        "Once you click the upload button, a file chooser dialog will open where you can select the `default_key.json`:\n",
+        "After the upload finished successfully, you should see the `default_key.json` in the file browser.\n",
+        "\n",
+        "![jupyterlab_service_key_uploaded_successfully.png](images/service_key_upload.png)\n",
```
The screenshots with the jupyter page and the notebook are still a bit outdated ("teched2020", and "key.json" are showing up there) - could we also update these? Btw, I find it quite nice to have the images as paths instead of attachments - they are still displayed in the browser (if you click on the notebook in github)... They are even visible in something like [mybinder.org](https://hub.gke2.mybinder.org/user/sap-data-attrib-tion-python-sdk-04i85m7y/lab/tree/notebooks/Data_Attribute_Recommendation_Generic_Model_Template.ipynb); with that, we can also share those same screenshots for the setup between the notebooks :)

karthikmswamy replied:
Thanks! Yes, I am also able to view the screenshots from the notebook on the branch. I've updated the branch to remove this screenshot as it looks like the text explains what is required and the screenshot was misleading. Hope this looks better.

misch commented on .gitignore:
```diff
@@ -130,3 +130,5 @@ system_test_results/
 env.yml
 
+# Keys
+notebooks/key.json
```
```suggestion
notebooks/default_key.json
```

misch reviewed (commented):
I really love the new content and explanations in the Regression Model notebook! Thank you for those updates! I've left only very minor comments

karthikmswamy commented:
Hi @misch, I've removed the misleading screenshot from all the notebooks. Let me know if this looks better.

karthikmswamy commented:
Hi @misch, I've updated the changelog to round up this PR. Hope this looks good now.

misch reviewed (approved):
Thanks @karthikmswamy, the notebooks really bring a lot of value, and this PR looks good to me now!

karthikmswamy commented:
Thanks for all your inputs, @misch :)
